rama-net/src/fingerprint/mod.rs

```diff
@@ -14,3 +14,111 @@ mod ja3;
 
 #[cfg(feature = "tls")]
 pub use ja3::{Ja3, Ja3ComputeError};
+
+#[cfg(feature = "tls")]
+mod tls_utils {
+    use private::ClientHelloProviderPriv;
+
+    /// Sealed trait used by ja3/ja4 computation functions,
+    /// to allow you to immediately compute from either a
+    /// [`ClientHello`] or a [`ClientConfig`] data structure.
+    ///
+    /// [`ClientHello`]: crate::tls::client::ClientHello
+    /// [`ClientConfig`]: crate::tls::client::ClientConfig
+    pub trait ClientHelloProvider: ClientHelloProviderPriv {}
+    impl<P: ClientHelloProviderPriv> ClientHelloProvider for P {}
+
+    mod private {
+        use crate::tls::{
+            CipherSuite, ProtocolVersion,
+            client::{ClientConfig, ClientHello, ClientHelloExtension},
+        };
+
+        pub trait ClientHelloProviderPriv {
+            fn protocol_version(&self) -> ProtocolVersion;
+            fn cipher_suites(&self) -> impl Iterator<Item = CipherSuite>;
```

Comment:
```suggestion
            fn cipher_suites(&self) -> impl Iterator<Item = &CipherSuite>;
```
More flexible and no need to copy here. In ja fingerprint logic you will be filtering first so copy after that makes more sense. Honestly performance wise this should never matter, but it's more for consistency

Reply:
ah yea true, thanks